services/motion/builtin/move_request.go

```diff
@@ -92,6 +93,45 @@ func (mr *moveRequest) Plan(ctx context.Context) (state.PlanResponse, error) {
 	}
 	mr.planRequest.StartConfiguration = map[string][]referenceframe.Input{mr.kinematicBase.Kinematics().Name(): inputs}
 
+	existingGifs, err := mr.planRequest.WorldState.ObstaclesInWorldFrame(
+		mr.planRequest.FrameSystem,
+		referenceframe.StartPositions(mr.planRequest.FrameSystem),
+	)
+	if err != nil {
+		return state.PlanResponse{}, err
+	}
+
+	existingGeoms := []spatialmath.Geometry{}
+	for _, g := range existingGifs.Geometries() {
+		if strings.Contains(g.Label(), "_transientObstacle_") {
+			continue
+		}
+		existingGeoms = append(existingGeoms, g)
+	}
+	existingGifs = referenceframe.NewGeometriesInFrame(referenceframe.World, existingGeoms)
+
+	// get transient detections
+	gifs := []*referenceframe.GeometriesInFrame{}
+	for visSrvc, cameraNames := range mr.obstacleDetectors {
+		for _, camName := range cameraNames {
+			currentPosition, err := mr.kinematicBase.CurrentPosition(ctx)
```

Comment:
I'd move this outside of the loop for speed reasons.

Reply:
It's kind of a tradeoff between "what if getTransientDetections takes too long and currentPosition is out of date" and "what if the whole loop takes too long and the obstacle poses from the first iteration are out of date" but either way making this loop as fast as possible minimizes error.